query-service-impl/src/main/java/org/hypertrace/core/query/service/postgres/converters/DefaultColumnRequestConverter.java

```diff
@@ -138,46 +141,87 @@ private String convertFilterToString(
           builder.append(" ");
           builder.append(
               convertLiteralToString(
-                  convertMapLikeExpressionToLiterals(filter.getRhs()), paramsBuilder, context));
+                  convertMapLikeExpressionToLiterals(filter.getRhs()), paramsBuilder));
           break;
         default:
-          Expression rhs =
-              handleValueConversionForLiteralExpression(filter.getLhs(), filter.getRhs());
-          builder.append(lhs);
-          builder.append(" ");
-          builder.append(operator);
-          builder.append(" ");
-          builder.append(convertExpressionToString(rhs, paramsBuilder, executionContext, context));
+          if (isFilterForBytesColumnType(filter, context)) {
+            handleConversionForBytesExpression(
+                lhs, operator, filter.getRhs(), builder, paramsBuilder);
+          } else {
+            builder.append(lhs);
+            builder.append(" ");
+            builder.append(operator);
+            builder.append(" ");
+            builder.append(
+                convertExpressionToString(
+                    filter.getRhs(), paramsBuilder, executionContext, context));
+          }
       }
     }
     return builder.toString();
   }
 
-  /**
-   * Handles value conversion of a literal expression based on its associated column.
-   *
-   * @param lhs LHS expression with which literal is associated with
-   * @param rhs RHS expression which needs value conversion if its a literal expression
-   * @return newly created literal {@link Expression} of rhs if converted else the same one.
-   */
-  private Expression handleValueConversionForLiteralExpression(Expression lhs, Expression rhs) {
-    if (!(isSimpleAttributeExpression(lhs) && rhs.getValueCase().equals(LITERAL))) {
-      return rhs;
+  private boolean isFilterForBytesColumnType(Filter filter, ColumnRequestContext context) {
+    return isSimpleAttributeExpression(filter.getLhs())
+        && filter.getRhs().getValueCase().equals(LITERAL)
+        && context.isBytesColumnType();
+  }
+
+  /** Handles value conversion of a bytes expression based */
+  private void handleConversionForBytesExpression(
+      String lhs, String operator, Expression rhs, StringBuilder builder, Builder paramsBuilder) {
+    Value value = rhs.getLiteral().getValue();
+    if (value.getValueType().equals(ValueType.STRING)) {
```

Comment:
nit: can we move to a separate function 
`handleEqualAndNotEqualOperatorForByteExpression`?